### ISSUE TITLE
Move and rename the `tokenise()` function for splitting strings to `core/foundation`

### ISSUE THREAD

**Issue by guitargeek** [root-project/root#8807]
In different ROOT components, it is convenient to have a helper function that takes a string, similar to the `str.split` function that Python has for example:

```Python
"a,b,c,d,e".split(",") # returns ['a', 'b', 'c', 'd', 'e']
```

Currently, there is code duplication to provide such a function in separate ROOT components like `roofit` and `hist`:
* https://github.com/root-project/root/blob/dcf920904564bbdca08a6acfa70019bbc24b3897/hist/hist/src/TFormula.cxx#L1678
* https://github.com/root-project/root/blob/dcf920904564bbdca08a6acfa70019bbc24b3897/roofit/roofitcore/src/RooHelpers.cxx#L67

As suggested by @Axel-Naumann in https://github.com/root-project/root/pull/8801#discussion_r683437844, this general function should be moved into `foundation/core` to about the code duplication and make the string-splitting functionality available the every ROOT component.

It is still an open question what the name of the function should be (as well as the name of the file it's in), and whether it should return a `std::vector<std::string>` or `std::vector<std::string_view>`.

**Comment by ferdymercury** [comment]
Side note: there is already https://root.cern/root/html530/TString.html#TString:Tokenize in core/base

although it is a bit tiring to work with:
```
TObjArray* tx = TString("a,b,c").Tokenize(",");
for (Int_t i = 0; i < tx->GetEntries(); i++) std::cout << ((TObjString *)(tx->At(i)))->String() << std::endl;
delete tx;
```

**Comment by Axel-Naumann** [comment]
Indeed, and a follow-up PR might change `TString::Tokenize()` to use the new facility here.

**Comment by eguiraud** [comment]
Another instance of a string splitting implementation is at https://github.com/root-project/root/blob/68c7ac93e0504dbd63075ede00a42eec131f4508/tree/ntuple/v7/src/RNTupleModel.cxx#L85-L95

**Comment by eguiraud** [comment]
About naming: although `std::strtok` does a simple string splitting, tokenization is often understood as a more complex operation than string splitting, involving some lexing, see e.g. https://docs.python.org/3/library/tokenize.html (vs the simple Python string split mentioned above) and https://github.com/ArashPartow/lexertk .

I would suggest to just call this `Split`, if other people share this "tokenizing sounds more complex than just splitting" feeling.

**Comment by eguiraud** [comment]
> whether it should return a std::vector<std::string> or std::vector<std::string_view>.

I'd say definitely `vector<string>`, otherwise calling things like `split(std::move(mystr))` or `split("a temporary str"s)` is broken or impractical. Thanks to short string optimization, creating several small strings is cheap.

**Comment by guitargeek** [comment]
Thanks for the discussion! So we go for `vector<string> ROOT::Split(const std::string &str, const std::string &delims, bool returnEmptyToken)` then? In which file should it go, maybe `ROOT/RSplit.hxx`? I think it would be good to also decide this before anyone can pick up this issue.

**Comment by Axel-Naumann** [comment]
IIRC we have a couple of other string functions that we'd really love to have, generally speaking. So maybe `ROOT/StringUtils.hxx`?

**Comment by guitargeek** [comment]
Sounds good to me!

**Comment by guitargeek** [comment]
Since the original `tokenise` function was used in RooFit a lot, I took care now of implementing the new `ROOT::Split` function.
